### ISSUE TITLE
Identical pressure between different sealed compartments / pressure zones without mass/heat transport

### ISSUE THREAD

**Issue by KingCZE** [firemodels/fds#11430]
Hello, I probably miss some understanding of how FDS works with pressure if not in regards to the mass and temperature. 

I created a simplified example (attached) from a case. Here we have two compartments, 3 zones (0-outer, 1-apartment/fire, 2-staircase). There is no leakage between Zone 1 and Zone 2 until a heat detector opens a duct dumper under the zone, and the pressure zones are sealed from each other (no error at the beginning of the simulation). The smoke / mass does not go between the zones until the duct has been activated. Yet, Zone 2 (staircase) copies the pressure measuring (in the HRR file) in Zone 1 (apartment/fire) from the very beginning of the simulation (without mass transport) and once the window in Zone 2 (staircase) opens, the pressure in Zone 1 (apartment/fire) drops as well (even without an opened duct). 

![image](https://user-images.githubusercontent.com/16994453/217395228-bd26f739-6fbf-4770-bee3-96ed6d4f78f0.png)

How does Zone 2 get increasing pressure (perfectly corresponding to Zone 1) when there is no mass or some extensive heat transport to it? What is the excessive mass in Zone 2 and where does it come from? 

![image](https://user-images.githubusercontent.com/16994453/217394223-bac78bc5-4acb-473e-874b-48bc3a2a8ae3.png)

The staircase window on the upper floor opens at the pressure peak, then the pressure drops. The pressure drops also in Zone 1 (apartment/fire); even though, there is no mass transport between them. The smoke is still kept in Zone 1, nothing leaks to Zone 2. 

Mass transport through the duct indeed activates when the damp on the duct is triggered (much later in the simulation).

I have tried ULMAT and it still leaks, so I doubt it is a leak through thin obstructions (and there are none), and even if there was some, the pressure still should not be identical.

[PRESSURE_ZONE_LEAK.fds.txt](https://github.com/firemodels/fds/files/10680479/PRESSURE_ZONE_LEAK.fds.txt)

**Comment by drjfloyd** [comment]
Thanks we will take a look at this.

**Comment by drjfloyd** [comment]
Kevin, Error still there if I simplify meshes down to this:
&MESH  IJK=24,67,18, XB=3.6,7.2,-0.9,9.15,0,2.7/R1+R2
&MESH  IJK=60,134,36, XB=-0.9,3.6,-0.9,9.15,0,2.7/
&MESH  IJK=20,68,20, XB=7.2,10.2,-0.9,9.3,0,3/
&MESH  IJK=20,76,32, XB=7.2,10.2,-0.9,10.5,3,7.8/

Plus up the VENT size so it appears on the grid for the HVAC vents:
&VENT ID='LOWER_DOOR_APT_HVAC_IN', SURF_ID='HVAC', XB=7.05,7.05,2.4,2.85,0.0,0.15/
&VENT ID='LOWER_DOOR_APT_HVAC_OUT', SURF_ID='HVAC', XB=7.2,7.2,2.4,2.85,0.0,0.15/

**Comment by drjfloyd** [comment]
Still there with the above changes and getting rid of LEAK_AREA, LEAK_PATH, and &HVAC.

**Comment by mcgratta** [comment]
[leak.txt](https://github.com/firemodels/fds/files/10688171/leak.txt)
Your case is not what I would consider simplified, and I have no idea what is wrong with it. Take the case that I have attached, and add to it the simplest possible version of your HVAC system, one component at a time. Do not add all the controls and devices. Just the basics. We need to slowly build the case from scratch and test the components one by one. 

The attached case is just the basic geometry and a blower in zone 1. The pressure rises in zone 1 only.

**Comment by drjfloyd** [comment]
I think I have determined the issue. It is this obstruct (I changed the color to make it clear which one it was).  This is one-half the door connecting the two sides. There is another pair of OBST that is the entire door. At t=0 the CTRL for this is FALSE. If I comment out the CTRL the two pressure zones diverge. I think somehow the pressure update is thinking the zones are joined because this OBST is not there  even though another OBST in the same location is.  Now that I think I know what the issue is I'll try and make a much simpler input.

&OBST ID='LOWER_DOOR_APT_OPEN-CLOSED', XB=7.05,7.2,2.85,3.3,0.0,2.1, PERMIT_HOLE=.FALSE., SURF_ID='DOOR', CTRL_ID='LOWER_DOOR_APT_CLOSE',COLOR='PURPLE'/

**Comment by drjfloyd** [comment]
Was able to simplify down to a couple dozen lines of input. With two meshes, this case shows the same pressure rise in each zone. WIth one mesh it doesn't.  There are two overlapping OBST one of which is not there at t=0.  Only the second OBST should be processed but with two meshes, this is not the case. The wall is solid for defining pressure zones but the pressure rise seems to treat it like it isn't there.

&HEAD CHID='leak'/
&TIME T_END=1.0/

!&MESH IJK=40,21,18, XB=4.2,10.2,1.8,4.95,0.0,2.7/1-mesh
&MESH IJK=20,21,18, XB=4.2,7.2,1.8,4.95,0.0,2.7/2-mesh
&MESH IJK=20,21,18, XB=7.2,10.2,1.8,4.95,0.0,2.7/2-mesh

&DEVC QUANTITY='TIME',SETPOINT=20,XYZ=7,3,1,ID='DEVC'/

&REAC FUEL='PROPANE'/
&SURF ID='FIRE',HRRPUA=500,COLOR='ORANGE'/

&VENT XB=4.5,4.8,2.1,2.4,0.0,0.0,SURF_ID='FIRE'/

&OBST XB=7.05,7.2,3.3,4.95,0.0,2.7/
&OBST XB=7.05,7.2,1.8,2.85,0.0,2.7/
&OBST XB=7.05,7.2,2.85,3.3,2.1,2.7/
&OBST XB=7.05,7.2,2.85,3.3,0.0,2.1, DEVC_ID='DEVC',COLOR='PURPLE'/
&OBST XB=7.05,7.2,2.85,3.3,0.0,2.1, COLOR='RED'/

&SLCF QUANTITY='PRESSURE', PBZ=2/
&SLCF QUANTITY='PRESSURE ZONE', PBZ=2/

&TAIL /

**Comment by mcgratta** [comment]
I set the `INITIAL_STATE` of the `DEVC` to T and the case works.  I think what is happening is that FDS is opening up the passage, including both blockages. Smokeview is not getting that message. I think we should somehow throw an error statement because we have two identical OBSTs with different properties.

**Comment by drjfloyd** [comment]
The User's Guide says that overlapping OBST the second takes priority. The second in this case is the closed door. FDS shouldn't be doing anything with the first OBST.  If I make the second OBST wider, this case still fails. I don't think we want to be throwing errors in the case of some overlap. That will cause havoc.

**Comment by mcgratta** [comment]
I think that the User's Guide is outdated. I don't think we reject over-lapping OBSTs anymore. I think what is happening is that whichever door is removed at the start, the cells are made non-SOILID.

**Comment by drjfloyd** [comment]
So we need some error trapping for when there is overlap and a difference in control logic?

**Comment by mcgratta** [comment]
I might have removed that check years ago when I discovered that it made reading in large cases really slow. Is the problem you identified the cause the original issue, or something you found in looking at the case?

**Comment by drjfloyd** [comment]
In the original issue, the problem was two OBST with the same XB were defined for the same location where one had a CTRL and the other did not.  FDS was removing based on the CTRL but Smokeview was showing the OBST. The ZONE search algorithm considers it present when defining ZONEs, but at t=0 it was removed. The guide isn't clear on how this is treated. In this specific case I think an error in making the input file in having two OBST with idenitcal XB defined that have different CTRL set for them. In a more general case though we could have two OBST with different XB where due to grid snapping or other cause we could have the case an OBST with a DEVC or CTRL uninentionally overlaps and OBST with no DEVC or CTRL and cause two ZONEs to merge that shouldn't.  When OBST overlap with DEVC or CTRL differences, I think we should have a warning or error that gives the OBST numbers and the cell location(s) like we do with VENTs that overlap.

**Comment by KingCZE** [comment]
Hello, any updates on the issue? 

As I understand, I will have to start using a different (more correct) way how to activate -> deactivate -> reactivate an obstruction. 

However, I find the issue quite serious when using the zones and all the functions associated with them. Particularly, if you use software such as Pyrosim etc., they sometimes generate a code with coordinates slightly out of grid (although usually some insanely low decimals), which then results in "overlapping" structures.

**Comment by mcgratta** [comment]
I think we need to return to the old way of rejecting over-lapping OBSTs. In the meantime, submit your input file to Thunderhead and ask how it came to be that two obstructions were put in the same place.

**Comment by KingCZE** [comment]
It is that one OBST deactivates at some conditions, and another one appears back later on some other conditions (with a delay).

**Comment by drjfloyd** [comment]
In this case there were two OBST that defined the door between sides. One had a control function and the other did not. FDS was using the OBST with the control function and that function was such that the OBST was not there at T_BEGIN. Smokeview gets a list of all OBST in the input file and draws all of them. So in Smokeview it didn't draw that OBST that was missing but it did draw the other OBST that FDS wasn't using. This made it look like the door was closed. Dealing with overlapping OBST has been a challenge especially if control functions are involved.

I think what we are doing in FDS now is OK as long as there are no control functions invovled. I think we should do something to throw an error or reject an OBST if there is overlap and a difference in the CTRL_ID or DEVC_ID. If there were overlapping OBST but the same control, this would have been fine.

This could be challenging to deal with in Pyrosim. If you are pulling in a large REVIT file and set THICKEN=.TRUE. for some or all layers, you can easily wind up with a multitude of overlapping OBST.

**Comment by KingCZE** [comment]
If it is just Smokeview, but FDS considers it opened, how is it that no smoke goes through to the other compartment?

**Comment by drjfloyd** [comment]
Sorry forgot some of the details of this case. FDS considered the OBST with the CTRL when considering what the pressure zones were. This opened the entire volume to the background pressure rise. The other OBST prevented flow. The process of adding/removing OBST and their impact on pressure zones is tricky logic. Generally if OBST overlap it is that they are always there so this issue wouldn't occur.

**Comment by mcgratta** [comment]
I looked again at the code and found that the test for `EMBEDDED` obstructions is still there. However, this check did not consider obstructions of the same size---only those that were completely within another. I changed that. Also, I added some logic to distinguish between obstructions under device control and those that are not. This means that the original case now stops with an error, and Jason's simpler test case simply rejects the last obstruction.
